flake8_fine_pytest/watchers/usefixtures.py

```diff
@@ -14,24 +20,62 @@ def run(self) -> None:
         if self._should_run():
             self._validate_usefixtures_used_where_possible(self.tree)
 
-    def _validate_usefixtures_used_where_possible(self, tree: ast.AST) -> None:
-        for node in ast.walk(tree):
-            if not self._should_check_node(node):
-                continue
+    def _should_run(self) -> bool:
+        return self.options.force_usefixtures and super()._should_run()
 
-            fixture_names = self._get_unreferenced_fixture_names(node)  # type: ignore
+    def _validate_usefixtures_used_where_possible(self, tree: ast.AST) -> None:
+        for test_node, is_class_member in self._iterate_over_test_function_definitions(tree):
+            fixture_names = self._get_unreferenced_fixture_names(
+                test_node, is_class_member,
+            )
             if fixture_names:
-                self._add_usefixtures_error(node, fixture_names)  # type: ignore
+                self._add_usefixtures_error(test_node, fixture_names)
+
+    def _iterate_over_test_function_definitions(
+        self, tree: ast.AST,
+    ) -> typing.Iterator[typing.Tuple[ast.FunctionDef, bool]]:
+        """
+        Returns any FunctionDef that looks like a test.
+
+        As pytest only discovers first-level-of-nesting tests, we only yield
+        top-level function definitions and top-level classes method definitions.
+        """
+        for node in ast.iter_child_nodes(self.tree):
+            if self._should_check_node(node):
+                yield node, False  # type: ignore
+
+            elif(
+                isinstance(node, ast.ClassDef)
```

Comment:
Let's move it to helper, or at least to variable with understandable name.

Reply:
Actually, we don't need this type of check

https://github.com/best-doctor/flake8-fine-pytest/pull/9/files#r533142561